src/reporting/branch.se

```diff
@@ -139,3 +139,13 @@ def isContainerForReportingToken(shadyReportingToken: address):
 def isParentOf(shadyChild: address):
     branchId = shadyChild.getParentPayoutDistributionHash()
     return(self.childBranches[branchId] == shadyChild)
+
+def isContainerForShareToken(denominationToken):
+    denominationTokenMarket = denominationToken.getMarket()
+    if (not denominationTokenMarket):
+        return(0)
+    if (not denominationTokenMarket.getBranch()):
+        return(0)
+    if (not denominationToken.isShareToken()):
+        return(0)
```

Comment:
This function needs to _trustlessly_ check to see if the denomination token is a shareToken.  At the moment, this code trusts what the `denominationToken` claims, though it could be malicious.

The way to do this is to ask the denomination token what "path" it claims to be at relative to the branch and then walk from the branch down that path and see if the provided path lands at `denominationToken`.

In this case, since we already have a function for `isContainerForMarket` you can just do something like:
```
shadyMarket = denominationToken.getMarket()
if not shadyMarket:
    return(0)
if not self.isContainerForMarket(shadyMarket):
    return(0)
if not market.isContainerForShareToken(shadyToken):
    return(0)
return(1)
```

You will need to add a function to `market` for `isContainerForShareToken(shadyShareToken)`.  This function would look something like:
```
outcome = shadyShareToken.getOutcome()
if self.getShareToken(outcome) == shadyShareToken:
    return(1)
else
    return(0)
```